app/src/features/mocks/screens/mocksList/components/MocksList/utils/index.ts

```diff
@@ -0,0 +1,26 @@
+import { MockTableHeaderFilter, RQMockMetadataSchema } from "components/features/mocksV2/types";
+import { isRecordMock, isRecordMockCollection } from "../components/MocksTable/utils";
+
+export const getQuickFilteredRecords = (records: RQMockMetadataSchema[], filterType: MockTableHeaderFilter) => {
+  switch (filterType) {
+    case "all": {
+      return records;
+    }
+    case "starred": {
+      const starredMocksCollectionIds: Set<string> = new Set();
+
+      records.forEach((record) => {
+        if (isRecordMock(record) && record.isFavourite && record.collectionId) {
+          starredMocksCollectionIds.add(record.collectionId);
+        }
+      });
+
+      return records.filter((record) => {
+        return isRecordMockCollection(record) ? starredMocksCollectionIds.has(record.id) : record.isFavourite;
+      });
```

Comment:
Move this enhancing logic to `MocksTable`. 

This might be a little bit slow but both the filtering and enhancing logics will be separate